### PR TITLE
Source to site calculation moved from IM_calculation repo

### PR DESCRIPTION
**Pull request by sungeunbae**
See https://github.com/ucgmsim/IM_calculation/pull/170

A PR from Empirical engine depends on this PR: https://github.com/ucgmsim/Empirical_Engine/pull/113

I am not entirely sure what to do with calculate_rrups_single.py - I think it is no longer needed to exist as a separate script.
Your feedback is welcome :)

### CODE REVIEW

File: qcore/source_site_dist/calculate_rrups_single.py
Comment (claudio525) [inline qcore/source_site_dist/calculate_rrups_single.py:1]: If we are not using it, then remove it I recon

Comment (sungeunbae) [reply qcore/source_site_dist/calculate_rrups_single.py:1]: Removed it

File: qcore/test/test_source_site_dist/test_src_site_dist_calc.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import numpy as np
+import pytest
+
+from qcore import geo, src_site_dist as ssd
+
+INPUT = "input"
+OUTPUT = "output"
+
+
+def test_calc_rrub_rjb(set_up):
```
Comment (joelridden) [inline qcore/test/test_source_site_dist/test_src_site_dist_calc.py:13]: The issues your having from the tests stem from this set_up which was a shared function for the tests in IM_calculation which setup the testing env with input output I believe

File: qcore/src_site_dist.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    else:
+        return rrups, rjb
+
+
+def calc_rx_ry(
+    srf_points: np.ndarray,
+    plane_infos: List[Dict],
+    locations: np.ndarray,
+    hypocentre_origin=False,
+    type=2,
```
Comment (lispandfound) [inline qcore/src_site_dist.py:84]: Missing type signatures

File: qcore/src_site_dist.py [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+    """
+    A wrapper script allowing external function calls to resolve to the correct location
+    :param srf_points: An array with shape (n, 3) giving the lon, lat, depth location of each subfault
+    :param plane_infos: A list of srf header dictionaries, as retrieved from qcore.srf.get_headers with idx=True
+    :param locations: An array with shape (m, 2) giving the lon, lat locations of each location to get Rx, Ry values for
+    :param type: Allows switching between the two GC types if desired
+    :param hypocentre_origin: If True sets the Ry origin/0 point to the fault trace projection of the hypocentre. If
+    false the most upstrike subfault of the first fault trace is used. Only used for GC2.
+    :return: An array with shape (m, 2) giving the Rx, Ry values for each of the given locations
+    """
```
Comment (lispandfound) [inline qcore/src_site_dist.py:95]: These docs should be in the numpydoc format.

File: qcore/src_site_dist.py [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+        )
+
+    return r_x, r_y
+
+
+def calc_rx_ry_GC2(
+    srf_points: np.ndarray,
+    plane_infos: List[Dict],
+    locations: np.ndarray,
+    hypocentre_origin=False,
```
Comment (lispandfound) [inline qcore/src_site_dist.py:186]: missing type signatures.

**Review by lispandfound — CHANGES_REQUESTED** [review]

File: qcore/src_site_dist.py [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+        )
+    else:
+        raise ValueError(f"Invalid GC type. {type} not in {{1,2}}")
+
+
+def calc_rx_ry_GC1(
+    srf_points: np.ndarray, plane_infos: List[Dict], locations: np.ndarray
+):
+    """
+    Calculates Rx and Ry distances using the cross track and along track distance calculations
```
Comment (AndrewRidden-Harper) [inline qcore/src_site_dist.py:110]: Echoing @lispandfound 's comment, this doc is also not in numpy format

File: qcore/src_site_dist.py [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+    locations: np.ndarray,
+    hypocentre_origin=False,
+):
+    """
+    Calculates Rx and Ry distances using the cross track and along track distance calculations
+    If there are multiple fault planes the Rx, Ry values are calculated for each fault plane individually, then weighted
+    according to plane length and distance to the location
+    For one fault plane this is the same as the GC1 function
+    Wrapper that calls calc_rx_ry_GC2_multi_hypocentre
+    :param srf_points: An array with shape (n, 3) giving the lon, lat, depth location of each subfault
```
Comment (AndrewRidden-Harper) [inline qcore/src_site_dist.py:194]: Also not in numpy doc format

**Review by AndrewRidden-Harper — COMMENTED** [review]
looks good. A few small comments

**Review by lispandfound — DISMISSED** [review]

File: qcore/test/test_source_site_dist/test_src_site_dist_calc.py [hunk 7]
```diff
@@ -1,13 +1,64 @@
...
+        unzip_cmd = "unzip {} -d {}".format(zip_download_path, data_store_path)
+        # print(DATA_STORE_PATH)
+        test_data_save_dirs.append(data_store_path)
+        if not os.path.isdir(data_store_path):
+            os.makedirs(data_store_path, exist_ok=True)
+            out, err = shared.exe(download_cmd, debug=False)
+            if "error" in err:
+                shutil.rmtree(data_store_path)
+                sys.exit("{} failed to retrieve test data".format(err))
+            # download_via_ftp(DATA_DOWNLOAD_PATH, zip_download_path)
```
Comment (lispandfound) [inline qcore/test/test_source_site_dist/test_src_site_dist_calc.py:39]: do we really need the commented lines here? We can always fish them out of the git history if we need them.

**Review by joelridden — APPROVED** [review]

**Comment by sungeunbae** [comment]
@joelridden @lispandfound @AndrewRidden-Harper   I just removed the test to get this PR merged quickly. Will add the test as a separate PR, I promise!

**Review by AndrewRidden-Harper — APPROVED** [review]

**Review by lispandfound — APPROVED** [review]

**Pull request merged**